pywr-schema/src/nodes/river.rs

```diff
@@ -31,17 +75,82 @@ impl RiverNode {
 
 #[cfg(feature = "core")]
 impl RiverNode {
+    /// The sub-name of the output node.
+    fn loss_node_sub_name() -> Option<&'static str> {
+        Some("loss")
+    }
+
     pub fn node_indices_for_constraints(
         &self,
         network: &pywr_core::network::Network,
     ) -> Result<Vec<pywr_core::node::NodeIndex>, SchemaError> {
         let idx = network.get_node_index_by_name(self.meta.name.as_str(), None)?;
         Ok(vec![idx])
     }
+
     pub fn add_to_model(&self, network: &mut pywr_core::network::Network) -> Result<(), SchemaError> {
         network.add_link_node(self.meta.name.as_str(), None)?;
+
+        // add output node and edge
+        if self.loss.is_some() {
+            network.add_output_node(self.meta.name.as_str(), Self::loss_node_sub_name())?;
+
+            let river = network.get_node_index_by_name(self.meta.name.as_str(), None)?;
```

Comment:
You could get these indices from the calls to `add_link_node` and `add_output_node` respectively.